extensions/ql-vscode/src/view/data-flow-paths/DataFlowPaths.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import styled from "styled-components";
+import { useState } from "react";
+
+import { useTelemetryOnChange } from "../common/telemetry";
+import { CodeFlowsDropdown } from "../common/CodePaths/CodeFlowsDropdown";
+import { SectionTitle, VerticalSpace } from "../common";
+import { CodePath } from "../common/CodePaths/CodePath";
+import { DataFlowPaths as DataFlowPathsDomainModel } from "../../variant-analysis/shared/data-flow-paths";
+
+const PathsContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const PathDetailsContainer = styled.div`
+  padding: 0;
+  border: 0;
+`;
+
+const PathDropdownContainer = styled.div`
+  flex-grow: 1;
+  padding: 0 0 0 0.2em;
+  border: none;
+`;
+
+export type DataFlowPathsProps = {
+  dataFlowPaths: DataFlowPathsDomainModel;
+};
+
+export const DataFlowPaths = ({
+  dataFlowPaths,
+}: {
+  dataFlowPaths: DataFlowPathsDomainModel;
+}): JSX.Element => {
+  const [selectedCodeFlow, setSelectedCodeFlow] = useState(
+    dataFlowPaths.codeFlows[0],
+  );
+  useTelemetryOnChange(selectedCodeFlow, "code-flow-selected");
+
+  const { codeFlows, ruleDescription, message, severity } = dataFlowPaths;
+
+  return (
+    <>
+      <SectionTitle>{ruleDescription}</SectionTitle>
```

Comment:
Should there be some extra whitespace above this text? It seems very close to the top of the webview.